modules/500-upmeter/hooks/dynamic_probe/hook.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2021 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamic_probe
+
+import (
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/yaml"
+
+	"github.com/deckhouse/deckhouse/go_lib/set"
+)
+
+// This hook populates internal values with object names that are used for dynamic probes in upmeter,
+// and are rendered in its Helm templates.
+var _ = sdk.RegisterFunc(
+	&go_hook.HookConfig{
+		OnBeforeHelm: &go_hook.OrderedConfig{Order: 10},
```

Comment:
Why do we need OnBeforeHelm?

Reply:
IIRC thus I can gather these values before Helm templates are rendered. I need it this way.